src/RefreshPattern.h

```diff
@@ -72,7 +76,28 @@ class RefreshPattern
         uint64_t matchCount;
         // TODO: some stats to indicate how useful/less the flags are would be nice.
     } stats;
+
+    /// reports configuration excluding trailing options
+    void printHead(std::ostream &) const;
+
+    /// reports the configured pattern or a fake pattern of the implicit rule
+    void printPattern(std::ostream &os) const;
+
+    // TODO: Refactor external refresh_pattern rules iterators to make private.
+    /// configured regex; do not use except when iterating configured rules
```

Comment:
The correct refactoring here would probably _not_ result in any friend functions, but that refactoring is quite complex and is outside this PR scope. That iterating code should be changed quite a bit; it is currently rather convoluted and fragile. This PR does not change that, and such changes would be out of this PR scope: This PR is about hiding regcomp()-related RegexPattern internals inside RegexPattern. It is not about improving RefreshPattern code quality. I only touched RefreshPattern code that I had to touch for in-scoppe reasons, and I stayed away from any serious refactoring of out-of-scope code.

Reply:
Sorry if I wasn't clear. I meant just using "friend" to keep this addition out of the RefreshPattern public API, reducing the comment warnings and notes. eg RefreshPattern should be touched *less* in this PR scope.

Just a nit. I do not insist on this.

Reply:
Your change request was clear. I was talking about proper refactoring. As for a quick "friends" hack, I think that adding two friends and changing one static function into a global one (to make it a friend without triggering [problems](https://stackoverflow.com/a/40937193)) is about the same as making this method public because this method can only be used (publicly) by those iterating/friend functions -- there is just no other way to access configured RefreshPattern objects.

Thank you for not insisting on this change.